apps/ledger-live-mobile/src/screens/CustomImage/Step1Crop.tsx

```diff
@@ -0,0 +1,188 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useNavigation, useRoute } from "@react-navigation/native";
+import { Flex, Icons, InfiniteLoader } from "@ledgerhq/native-ui";
+import { CropView } from "react-native-image-crop-tools";
+import { useTranslation } from "react-i18next";
+import { Dimensions, Platform } from "react-native";
+import ImageCropper, {
+  Props as ImageCropperProps,
+  CropResult,
+} from "../../components/CustomImage/ImageCropper";
+import {
+  ImageDimensions,
+  ImageFileUri,
+  ImageUrl,
+} from "../../components/CustomImage/types";
+import {
+  fitImageContain,
+  loadImageToFileWithDimensions,
+} from "../../components/CustomImage/imageUtils";
+import { cropAspectRatio } from "./shared";
+import Button from "../../components/Button";
+import { ScreenName } from "../../const";
+import BottomContainer from "../../components/CustomImage/BottomButtonsContainer";
+import Touchable from "../../components/Touchable";
+
+type RouteParams = Partial<ImageUrl> &
+  Partial<ImageFileUri> &
+  Partial<ImageDimensions>;
+
+export const boxToFitDimensions = {
+  width: Dimensions.get("screen").width,
+  height: Dimensions.get("screen").height,
+};
+
+/**
+ * UI component that loads the input image (from the route params) &
+ * displays it in a cropping UI with a confirm button at the bottom.
+ * Then on confirmation it navigates to the preview step with the cropped image
+ * file URI as a param.
+ */
+const Step1Cropping: React.FC<{}> = () => {
+  const cropperRef = useRef<CropView>(null);
+  const [imageToCrop, setImageToCrop] = useState<
+    (ImageFileUri & Partial<ImageDimensions>) | null
+  >(null);
+  const [rotated, setRotated] = useState(false);
+
+  const { t } = useTranslation();
+
+  const navigation = useNavigation();
+  const { params }: { params: RouteParams } = useRoute();
```

Comment:
TS is complaining here - but i guess if you add types to the navigator it will stop complaining ? And i don't know what's best: `useNavigation` and `useRoute`, or pass `navigation` and `route` as props to your component (which is what i did [here](https://github.com/LedgerHQ/ledger-live/blob/aa76c034a6400add49b20d79aba9089681c235f7/apps/ledger-live-mobile/src/screens/SyncOnboarding/BleDevicePairing.tsx#L22) and the props type is defined using `StackScreenProps` [here](https://github.com/LedgerHQ/ledger-live/blob/aa76c034a6400add49b20d79aba9089681c235f7/apps/ledger-live-mobile/src/screens/SyncOnboarding/BleDevicePairing.tsx#L17)) ? With this  props definition, i don't get TS complaining